### ISSUE TITLE
PCL target & NuGet package

### ISSUE THREAD

**Issue by varon** [logibit/Logibit.Hawk#9]
Would it be possible to get this targeting and releasing with a PCL build? It would be good to use the Hawk Client from Xamarin apps.

For the libraries, NodaTime is already supported as a PCL, so the only major work would be relatively minor adjustments.

This would be advantageous to any user wanting to consume an API secured with Hawk from Xamarin.

**Comment by haf** [comment]
Yes perhaps it works by now. I've been bitten by PCL before, but for you @varon I will give it a shot.

**Comment by haf** [comment]
It seems crypto isn't available. Using System.Cryptography breaks. Googling says I should use PCLCrypto but that just supports Windows so what's the point? https://github.com/AArnott/PCLCrypto/issues/103

**Comment by haf** [comment]
Also missing a memory cache that times out https://stackoverflow.com/questions/20160014/is-there-a-in-memory-cache-for-portable-class-library

**Comment by haf** [comment]
And it would seem we're missing a Http Client too

**Comment by haf** [comment]
@varon Would you consider PRing the basics? I can help out.

**Comment by varon** [comment]
@haf  Thanks for investigating this.

PCLCrypto has NuGet packages for just about everything, including WP and Xamarin (iOS and Android). Not sure about .Net Core, but at least we could use Hawk client from the mobile apps through Xamarin.

MemoryCache I just turned off, i.e. had it default to the Choice2Of2 to get it to compile, the idea was that it could be replaced with our own implementation. I'm not sure how important this is, but I'd imagine id should be pretty trivial to provide our own implementation.

HttpClient actually is supported on most platforms, but it's as a platform specific library.
We can either link to those, or we could try use the [Microsoft PCL](https://www.nuget.org/packages/Microsoft.Net.Http), although it's worth checking for potential licensing issues here.

For PRing the basics, I'm not actually in need of Hawk for any project at the moment, but was surprised to see that it didn't yet have support for this. If you're critically short on time, I can look at getting either myself or someone else onto this, but would prefer to use those resources elsewhere if possible.

The feedback and assistance so far is really appreciated.

**Comment by haf** [comment]
I'm quite time limited, to be honest. I do mostly server-side stuff too, so for me it would be .Net Core next – and that seems to use _yet another API_ (e.g. CultureInfo from Globalization is different) I'll leave the issue open for now.

**Comment by varon** [comment]
No trouble at all, thanks for accommodating.
